File: java/7- Sorting/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {
    public static void swap(int[] arr, int s, int e) {
        int temp = arr[s];
        arr[s] = arr[e];
        arr[e] = temp;

    }

    public static void main(String[] args) {

        int[] arr = { 3, 5, 4, 1, 2 };

        for (int i = 0; i < arr.length; i++) {
            boolean isSorted = false;
            for (int j = 1; j < arr.length - i; j++) {
                if (arr[j] < arr[j - 1]) {
                    swap(arr, j, j - 1);
                    isSorted = true;
                }
            }
            if (!isSorted) {
                break;
            }
        }

        System.out.println(Arrays.toString(arr));
    }
}

File: java/Learning Recursion/sorting questions/BubbleSort.py
myArr = [4,3,2,1]

def bubbleSort(arr,r,c):
    if r == 0:
        return
    
    if c < r:
        if arr[c] > arr[c + 1]:
            # swap 
            temp = arr[c]
            arr[c] = arr[c+1]
            arr[c+1]=temp

        bubbleSort(arr,r,c+1)
    else:
        bubbleSort(arr,r-1,0)

bubbleSort(myArr,len(myArr) - 1,0)
print(myArr)